### ISSUE TITLE
Proposal : Portable version of the benchmark

### ISSUE THREAD

**Issue by AlexandreBrown** [gemcollector/RL-ViGen#20]
# Context
Hi, I like what has been done with RL-ViGen and I like that this benchmark exposes well that most approaches are not perfect in all scenarios.  I think ideally, RL-ViGen should be used for future visual RL research.  
That being said, I find it fairly hard to integrate with existing repositories which makes it impossible to use for research unless one re-writes his code for the RL-ViGen repo. Even then, it's hard to get it to work depending on your linux version for instance and using docker is not great for quick R&D/Debugging.  
I really want to use RL-ViGen but given the complexity to integrate this might not be possible unless we get an option for a portable version.
    
# Proposal
I would like to propose the idea of a portable version of the benchmark that would only contain the environments.  
This would make it possible for people to simply create the environments and integrate it with their existing research repository which might already contain the instantiation of other environments, training code and more complex code that RL-ViGen doesn't have to maintain.  
Other benchmarks seem to follow a similar approach (eg: https://github.com/vikashplus/robohive  can be install via pip and we can create an env in a few lines of code).   
It would be great if RL-ViGen could do the same or at least offer the option to only need the environments (no algorithms or training code or pre-established hydra configs, envs would be created in python and accept a `dict`).   
This would also remove constraints like needing hydra, tensorboard, weight & biases or needing a specific PyTorch version (currently it is pinned in the repo which is far from ideal). 
    
# Ideal Solution  
I think the ideal solution would be to be able to pip install `RL-ViGen` and the package could contain the envs and offer create functions like `create_carla_envs` etc.  
### Prototype
I tried doing a prototype of another idea here : https://github.com/AlexandreBrown/RL-ViGen-Portable  
The idea is to allow someone to go from an existing conda environment from an existing project repository and simply do :  
1. Clone the portable repo
2. Install RL-ViGen dependencies: `conda env update --file environment.yaml`
3. Create the RL-ViGen environments in code, example : `from wrappers.habi_wrappers import make_habitat_env` then `env = make_habitat_env(...)`

The repository is still WIP and while the conda update works right now, I still get issues when trying to create the environments in python. The goal of the repository was just to give an idea of the proposal but not necessarily to be the complete solution so I hope it gives the gist of the idea.  The README contains additional details.   

This might be trivial for someone already familiar with RL-ViGen (ie: The authors) so I am taking the time to ask if this could be added in the future via a branch (eg: `envs-only`) or even better `pip install rl_vigen` or something like that.  
  
Thank you and feel free to give your feedback,

**Comment by AlexandreBrown** [comment]
@gemcollector Would love your feedback on this if possible :)

**Comment by gemcollector** [comment]
Hi @AlexandreBrown , Thank you for your detailed proposal and illustration. We are delighted to see the portable version of rl-vigen, which simplifies the installation process and usage. We are also very willing to maintain the environment part as a branch of your repo. We do appreciate your support for our work!

**Comment by AlexandreBrown** [comment]
@gemcollector Ok this is great, I just pushed a commit to the fork that makes Carla env work on Ubuntu 24.04 and I don't see why it wouldn't work on other versions as I also updated carla sim from 0.9.10 to 0.9.12 which now uses pip to install carla python api instead of a the egg file. I updated the carla env code to work with 0.9.12 as there were breaking changes.  
You can see a working minimal example here : https://github.com/AlexandreBrown/RL-ViGen-Portable/blob/master/doc/carla/README.md   
I also made the code more portable by not setting the PYTHONPATH in code but rather before running the code (so no hard coding of files paths).
  
I did not make the changes for the other environments yet but so far carla works.

**Comment by gemcollector** [comment]
How should I reproduce this error? Directly install your repo?

**Comment by AlexandreBrown** [comment]
@gemcollector Yes simply clone the repo, follow https://github.com/AlexandreBrown/RL-ViGen-Portable/tree/master?tab=readme-ov-file#portable-installation then after that for robosuite env creation, follow : https://github.com/AlexandreBrown/RL-ViGen-Portable/blob/master/doc/robosuite/README.md#prerequisites
